### ISSUE TITLE
"blik" context word "long/noun" registered as "darjoolong/noun"

### ISSUE THREAD

**Issue by AntheSevenants** [montesmariana/tokenclouds#1]
In [`blik.ppmi.tsv`](https://github.com/QLVL/tokenclouds/blob/main/data/blik/blik.ppmi.tsv), the context word "long/noun" has a weird id, "darjoolong/noun". This will cause NephoVis to crash. Maybe something is wrong in the data collection pipeline?

**Comment by montesmariana** [comment]
I don't think _darjoolong/noun_ is a transformation of _long/noun_.

From what I found inspecting `blik.variables.tsv`, there is only one token of _blik_ that co-occurs with _long/noun_, and then the context word is only captured in dependency-based models that extend to three steps in the dependency path and have no PPMI filter. In that token, the context word _long/noun_ occurs more than 15 words away of the target, so it's not even present in the raw context provided in `blik.variables.tsv`! (That _should_ be fixed in the pipeline, I used to cover the full sentence in a previous version.)
So, _long/noun_ probably never occurs in the 10-10 around _blik_ that was used to compute the selection and frequency in `blik.ppmi.tsv`. _darjoolong/noun_, on the other hand, might co-occur with _blik_ at some point in the corpus, although not in the modeled sample, and therefore ended up in the `ppmi.tsv` file, unnecessarily.

In short, I don't think there is a connection between _long/noun_ and _darjoolong/noun_ in `blik.ppmi.tsv`, just that one context word is missing from `blik.ppmi.tsv` and another one never shows up in `blik.variables.tsv`. In any case, why would that crash NephoVis?

**Comment by AntheSevenants** [comment]
Thanks for investigating! Strange how that works...

I selected a number of tokens and opened the frequency table, which then promptly crashed. The issue was that it couldn't find `long/noun` in the `ppmi` dataset, and we never account for this situation happening.

I've since [fixed](https://github.com/AntheSevenants/NephoVis/commit/66d2c1ddcce54e73bce4088f6f556db37f1fe251) the issue, but I just wanted to let you know, because it's of course never a good thing when there's missing data.

**Comment by montesmariana** [comment]
Indeed, it just didn't come up in the previous version (I think the table was filtered by the context words found but didn't check whether all the context words were included). In any case, this is up to the researcher who builds the `ppmi.tsv` file, I believe, and it's a specific situation of mixing dependency-based models that can go beyond the window size and BOW-based frequency counts. I will add a warning in the nephosem and semasioFlow docs. Thanks for finding it!
